### PR TITLE
[17.0][MIG] sale_order_line_position: Migration to 17.0

### PR DESCRIPTION
**Pull request by xaviedoanhduy**
[Order - S00024 (1).pdf](https://github.com/xaviedoanhduy/sale-reporting/files/15019247/Order.-.S00024.1.pdf)

### CODE REVIEW

File: sale_order_line_position/models/ir_actions_report.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class IrActionsReport(models.Model):
+    _inherit = "ir.actions.report"
+
+    def render_qweb_pdf(self, res_ids=None, data=None):
```
Comment (nguyenminhchien) [inline sale_order_line_position/models/ir_actions_report.py]: it should be `def _render_qweb_pdf()`. btw, you must make sure the positions are recomputed when printing the Quotation / Order report.

Comment (nguyenminhchien) [reply sale_order_line_position/models/ir_actions_report.py]: it's better if you create a unit test for this. so, we don't miss this point next time of the migration.

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]

File: sale_order_line_position/models/ir_actions_report.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    def _render_qweb_pdf(self, report_ref, res_ids=None, data=None):
+        self.sale_recompute_positions(res_ids)
+        return super()._render_qweb_pdf(report_ref, res_ids, data)
+
+    def sale_recompute_positions(self, res_ids):
+        context = self.env.context
+        active_model = context.get("active_model", "")
+
+        if active_model == "sale.order":
```
Comment (nguyenminhchien) [inline sale_order_line_position/models/ir_actions_report.py]:
```
        report_sudo = self._get_report(report_ref)
        if report_sudo.model == "sale.order":
            sales = self.env["sale.order"].browse(res_ids)
            sales.recompute_positions()
```

File: sale_order_line_position/static/description/index.html [hunk 3]
```diff
@@ -0,0 +1,458 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (nguyenminhchien) [inline sale_order_line_position/static/description/index.html]: remove this line

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]

**Comment by xaviedoanhduy** [comment]
Thanks for reviewing. The PR commit was changed accordingly.

**Review by nguyenminhchien — APPROVED** [review]

**Pull request closed without merging**